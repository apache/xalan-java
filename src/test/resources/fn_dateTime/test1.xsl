<?xml version="1.0"?> 
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                xmlns:xs="http://www.w3.org/2001/XMLSchema"
                exclude-result-prefixes="xs"
                version="3.0">
                
  <!-- Author: mukulg@apache.org -->
  
  <!-- An XSLT stylesheet test case, to test XPath 3.1 
       function fn:dateTime. 
  -->                  
                
  <xsl:output method="xml" indent="yes"/>    

  <xsl:variable name="dateTime1" select="dateTime(xs:date('1999-12-31'), xs:time('12:00:00'))"/>  
  <xsl:variable name="dateTime2" select="dateTime(xs:date('1999-12-31'), xs:time('24:00:00'))"/>
  <xsl:variable name="dateTime3" select="dateTime(xs:date('2012-10-05+05:30'), xs:time('24:00:00'))"/>
  
  <xsl:template match="/">
    <result>      
      <one isXsDateTimeValue="{$dateTime1 instance of xs:dateTime}">
        <xsl:value-of select="$dateTime1"/>
      </one>
      <two isXsDateTimeValue="{$dateTime2 instance of xs:dateTime}">
        <xsl:value-of select="$dateTime2"/>
      </two>
      <three isXsDateTimeValue="{$dateTime3 instance of xs:dateTime}">
        <xsl:value-of select="$dateTime3"/>
      </three>
    </result> 
  </xsl:template>
  
  <!--
      * Licensed to the Apache Software Foundation (ASF) under one
      * or more contributor license agreements. See the NOTICE file
      * distributed with this work for additional information
      * regarding copyright ownership. The ASF licenses this file
      * to you under the Apache License, Version 2.0 (the  "License");
      * you may not use this file except in compliance with the License.
      * You may obtain a copy of the License at
      *
      *     http://www.apache.org/licenses/LICENSE-2.0
      *
      * Unless required by applicable law or agreed to in writing, software
      * distributed under the License is distributed on an "AS IS" BASIS,
      * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      * See the License for the specific language governing permissions and
      * limitations under the License.
  -->

</xsl:stylesheet>