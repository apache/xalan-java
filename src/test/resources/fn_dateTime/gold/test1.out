<?xml version="1.0" encoding="UTF-8"?><result>
  <one isXsDateTimeValue="true">1999-12-31T12:00:00</one>
  <two isXsDateTimeValue="true">1999-12-31T00:00:00</two>
  <three isXsDateTimeValue="true">2012-10-05T00:00:00+05:30</three>
</result>
